### PR TITLE
Fix python3+osx tests on Travis

### PR DESCRIPTION
**Pull request by tkf**
Previously, osx on Travis runs Python 2 even when the intention was to run the tests with Python 3.  See:
https://travis-ci.org/JuliaPy/pyjulia/jobs/373733201#L256
https://github.com/JuliaPy/pyjulia/pull/149#issuecomment-389380351

This is likely because:

> `python` points to Homebrew’s Python 2.7.x (if installed) otherwise
> the macOS system Python.
> -- https://docs.brew.sh/Homebrew-and-Python

This patch forces tests to run with whatever specified by `PYTHON` environment variable.  This also let PyCall's build script to use the Python executable specified.

### CODE REVIEW

**Comment by tkf** [comment]
https://github.com/JuliaPy/pyjulia/pull/149#issuecomment-389756222

File: .travis.yml [hunk 1]
```diff
@@ -64,6 +64,8 @@ before_script:
   - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install python-numpy python3-numpy -y; fi
   - if [ "$TRAVIS_OS_NAME" = "osx" -a "$PYTHON" = "python3" ]; then brew update; brew list python &>/dev/null || brew upgrade python || echo "Ignoring errors..."; fi
   - if [ "$TRAVIS_OS_NAME" = "osx" -a "$PYTHON" = "python2" ]; then brew update; brew list python@2 &>/dev/null || brew install python@2 || echo "Ignoring errors..."; fi
+  - ls /usr/local/bin/py*
+  - for p in /usr/local/bin/python*; do echo $p; $p --version; done
```
Comment (tkf) [inline .travis.yml]: Note:

```
$ ls /usr/local/bin/py*
/usr/local/bin/pydoc2		/usr/local/bin/python2
/usr/local/bin/pydoc2.7		/usr/local/bin/python2-config
/usr/local/bin/pyenv		/usr/local/bin/python2.7
/usr/local/bin/pyenv-install	/usr/local/bin/python2.7-config
/usr/local/bin/pyenv-uninstall	/usr/local/bin/pythonw2
/usr/local/bin/python-build	/usr/local/bin/pythonw2.7
before_script.9
0.26s$ for p in /usr/local/bin/python*; do echo $p; $p --version; done
/usr/local/bin/python-build
python-build 20160602
/usr/local/bin/python2
Python 2.7.14
/usr/local/bin/python2-config
Usage: /usr/local/bin/python2-config [--prefix|--exec-prefix|--includes|--libs|--cflags|--ldflags|--help]
/usr/local/bin/python2.7
Python 2.7.14
/usr/local/bin/python2.7-config
Usage: /usr/local/bin/python2.7-config [--prefix|--exec-prefix|--includes|--libs|--cflags|--ldflags|--help]
/usr/local/bin/pythonw2
Python 2.7.14
/usr/local/bin/pythonw2.7
Python 2.7.14
```
https://travis-ci.org/JuliaPy/pyjulia/jobs/380142002#L2485

with https://github.com/tkf/pyjulia/commit/d0ae0642ecbe623946845657bf98efd4cc8a5b0d

**Comment by rdeits** [comment]
@tkf I now have access to this repo, so I'm going to go ahead and merge!

**Comment by tkf** [comment]
Thanks!

**Pull request merged**